tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
-# Distributed key generation gossip
+# Distributed key generation
+
+Here we give an overview of how the DKG protocol works with the ledger and 
+the roles of each node in the network per this protocol. This portion of the 
+book focuses solely on the production of a new public encryption key via the
+DKG scheme. It does not cover encryption, decryption, or validation of transactions.
+
+Some parts of the DKG protocol may be done through the gossip layer or through
+off-chain communication. However, the entire DKG cannot exist solely in the 
+gossip layer as consensus on the final shared secrets must be achieved. This 
+will require posting information on chain.
+
+## Starting the Protocol
+
+To initializing the protocol to generate a new key, the  current block 
+proposer proposes a new instance. This would be posted on the blockchain and
+notify all the participating validators to begin the announcement phase. 
+However, when should the proposer make this decision? In all likelihood it will be a combination of blocks transpired since
+the last instance completed as well as special cases (such as an instance
+failing to complete).
+
+## Announcing and distributing weight shares
+
+In order for the DKG protocol to work, it requires the contribution of 
+ \\(\frac{2}{3} \\) of the total staked weight among live validators. In 
+order to know when a subset of validators own at least \\(\frac{2}{3}\\)
+of the total weight, it is necessary for validators to announce their stake.
```

Comment:
This means that we could give out weight shares to non-live validators. This probably not a big concern, but in the current setup, since announcements must be proactive, we get a handle on who is live and who isn't. But if we assume liveness, we can make this change

Reply:
My concern is that liveness (or lack of liveness) should not propagate past the actual unlive period. For example, if a validator is unlucky enough to go offline during some of the DKG, they should not be locked out of validating the entire epoch - but if they fail to announce during the DKG, then they don't get access to their key shares and they're effectively jailed the entire time.

On one hand, sure we want to incentivize liveness, but I'm concerned this creates a denial-of-service attack window  where a short attack can affect the network for a longish period

Reply:
This is a good point. I agree with you. And generally we don't expect a lack of liveness to be a huge issue I think.